### PR TITLE
Fix a couple of pan zoom bugs

### PR DESCRIPTION
**Pull request by eatyourgreens**
Checks image height and width for changes before resetting zoom for a new subject.

Fixes some eslint warnings.

Adds an eslint rule for switch/case indentation.

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Did you deploy a staging branch? https://pan-zoom.pfe-preview.zooniverse.org/
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [x] Are the tests passing locally and on Travis?

## Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

File: app/components/pan-zoom.jsx [hunk 1]
```diff
@@ -126,16 +127,17 @@ class PanZoom extends React.Component {
...
   }
 
-  zoomReset() {
+  zoomReset(props) {
+    props = props || this.props;
     this.setState({
       viewBoxDimensions: {
-        width: this.props.frameDimensions.width,
-        height: this.props.frameDimensions.height,
+        width: props.frameDimensions.width,
```
Comment (srallen) [inline app/components/pan-zoom.jsx]: Testing with the current PRN project (https://pan-zoom.pfe-preview.zooniverse.org/projects/vrooje/planetary-response-network-and-rescue-global-caribbean-storms-2017/classify?env=production), if you rotate or zoom in the subject image and hit the reset button, it'll throw ` TypeError: Cannot read property 'width' of undefined`. I think the issue is this line, but it's a little hard for me to be sure looking at the minified file.

**Comment by eatyourgreens** [comment]
@srallen that's the right workflow. The issue showed up when you finished a classification and loaded the next subject, because the subjects have different aspect ratios and sizes. I used demo mode to reproduce it.

**Comment by srallen** [comment]
@eatyourgreens thanks for the clarification. I'll take a look after the zoominar.

File: app/components/pan-zoom.jsx (hunk 1 above)
Comment (eatyourgreens) [reply app/components/pan-zoom.jsx]: My bad. I broke the handler for that button.

Comment (eatyourgreens) [reply app/components/pan-zoom.jsx]: Should be fixed by f5066ed

**Comment by eatyourgreens** [comment]
Prop warnings are fixed for the tests too.

**Pull request merged**